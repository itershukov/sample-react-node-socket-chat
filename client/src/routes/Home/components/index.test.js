import React from 'react';
import ReactDOM from 'react-dom';
import View from './index';
import renderer from 'react-test-renderer';
import { mount, render, shallow } from 'enzyme';
import { Button, Menu, Form } from 'antd';
const FormItem = Form.Item;

describe(`Base test <Home />`, () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<View />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});

describe(`Snapshot <Home />`, () => {
  it('View should be the same', () => {
    const component = renderer.create(<View />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe(`<Home />`, () => {
  it('renders button', () => {
    const wrapper = mount(<View />);
    expect(wrapper.contains(Button)).toEqual(true);
  });

  it('renders only one button', () => {
    const wrapper = mount(<View />);
    const btns = wrapper.find(Button);
    expect(btns.length).toEqual(1);
  });

  it('renders header', () => {
    const wrapper = mount(<View />);
    expect(
      wrapper.contains(
        <p>Please enter your name as you want it to appear in the chat</p>
      )
    ).toEqual(true);
  });

  it('renders input', () => {
    const wrapper = mount(<View />);
    expect(wrapper.find(FormItem).length).toEqual(2);
  });

  it('logout is absent', () => {
    const wrapper = mount(<View />);
    expect(wrapper.find(Menu).length).toEqual(0);
  });

  it('submit empty form', () => {
    const wrapper = mount(<View />);

    let warning = wrapper.find('.ant-form-explain');

    expect(warning.length).toEqual(0);

    wrapper.find('button').simulate('submit');

    warning = wrapper.find('.ant-form-explain');

    expect(warning.length).toEqual(1);
  });
});
