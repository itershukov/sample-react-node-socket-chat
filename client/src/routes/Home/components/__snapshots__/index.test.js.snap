// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot <Home /> View should be the same 1`] = `
<div
  className="ant-row"
  style={undefined}
>
  <div
    className="ant-col-12 ant-col-offset-6"
  >
    <div
      className="ant-card ant-card-bordered"
      style={
        Object {
          "marginTop": 20,
        }
      }
    >
      <div
        className="ant-card-body"
        style={undefined}
      >
        <form
          className="ant-form ant-form-horizontal"
          onSubmit={[Function]}
        >
          <p>
            Please enter your name as you want it to appear in the chat
          </p>
          <div
            className="ant-row ant-form-item"
            style={undefined}
          >
            <div
              className="ant-form-item-control-wrapper"
            >
              <div
                className="ant-form-item-control"
              >
                <span
                  className="ant-form-item-children"
                >
                  <input
                    className="ant-input"
                    data-__field={
                      Object {
                        "name": "nickname",
                      }
                    }
                    data-__meta={
                      Object {
                        "name": "nickname",
                        "originalProps": Object {
                          "disabled": false,
                          "placeholder": "Nickname",
                          "prefixCls": "ant-input",
                          "type": "text",
                        },
                        "ref": null,
                        "rules": Array [
                          Object {
                            "message": "Please input your username!",
                            "required": true,
                          },
                        ],
                        "trigger": "onChange",
                        "validate": Array [
                          Object {
                            "rules": Array [
                              Object {
                                "message": "Please input your username!",
                                "required": true,
                              },
                            ],
                            "trigger": Array [
                              "onChange",
                            ],
                          },
                        ],
                        "valuePropName": "value",
                      }
                    }
                    disabled={false}
                    id="nickname"
                    onChange={[Function]}
                    onKeyDown={[Function]}
                    placeholder="Nickname"
                    type="text"
                    value=""
                  />
                </span>
              </div>
            </div>
          </div>
          <div
            className="ant-row ant-form-item"
            style={undefined}
          >
            <div
              className="ant-form-item-control-wrapper"
            >
              <div
                className="ant-form-item-control"
              >
                <span
                  className="ant-form-item-children"
                >
                  <button
                    className="ant-btn ant-btn-primary"
                    onClick={[Function]}
                    type="submit"
                  >
                    <span>
                      Start messaging
                    </span>
                  </button>
                </span>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`Snapshot <Home /> View should be the same 2`] = `
<div
  className="ant-row"
  style={undefined}
>
  <div
    className="ant-col-12 ant-col-offset-6"
  >
    <div
      className="ant-card ant-card-bordered"
      style={
        Object {
          "marginTop": 20,
        }
      }
    >
      <div
        className="ant-card-body"
        style={undefined}
      >
        <form
          className="ant-form ant-form-horizontal"
          onSubmit={[Function]}
        >
          <p>
            Please enter your name as you want it to appear in the chat
          </p>
          <div
            className="ant-row ant-form-item"
            style={undefined}
          >
            <div
              className="ant-form-item-control-wrapper"
            >
              <div
                className="ant-form-item-control"
              >
                <span
                  className="ant-form-item-children"
                >
                  <input
                    className="ant-input"
                    data-__field={
                      Object {
                        "name": "nickname",
                      }
                    }
                    data-__meta={
                      Object {
                        "name": "nickname",
                        "originalProps": Object {
                          "disabled": false,
                          "placeholder": "Nickname",
                          "prefixCls": "ant-input",
                          "type": "text",
                        },
                        "ref": null,
                        "rules": Array [
                          Object {
                            "message": "Please input your username!",
                            "required": true,
                          },
                        ],
                        "trigger": "onChange",
                        "validate": Array [
                          Object {
                            "rules": Array [
                              Object {
                                "message": "Please input your username!",
                                "required": true,
                              },
                            ],
                            "trigger": Array [
                              "onChange",
                            ],
                          },
                        ],
                        "valuePropName": "value",
                      }
                    }
                    disabled={false}
                    id="nickname"
                    onChange={[Function]}
                    onKeyDown={[Function]}
                    placeholder="Nickname"
                    type="text"
                    value=""
                  />
                </span>
              </div>
            </div>
          </div>
          <div
            className="ant-row ant-form-item"
            style={undefined}
          >
            <div
              className="ant-form-item-control-wrapper"
            >
              <div
                className="ant-form-item-control"
              >
                <span
                  className="ant-form-item-children"
                >
                  <button
                    className="ant-btn ant-btn-primary"
                    onClick={[Function]}
                    type="submit"
                  >
                    <span>
                      Start messaging
                    </span>
                  </button>
                </span>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
